pkg/netpol/connlist/connlist_test.go

```diff
@@ -1,44 +1,104 @@
+	expectedOutputFileName, testName, format string) {
+	analyzer, res, err := getConnlistFromDirPathRes(analyzerOptions, dirName)
+	require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+	output, err := analyzer.ConnectionsListToString(res)
+	require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+	checkActualVsExpectedOutputMatch(t, testName, dirName, expectedOutputFileName, output, format)
+}
+
+// helping func - writes debug message for good path tests
+func getDebugMsgWithTestNameAndFormat(testName, format string) string {
+	return fmt.Sprintf("test: %q, output format: %q", testName, format)
 }
 
-const expectedOutputFileNamePrefix = "connlist_output."
+// helping func - checks if actual output matches expected output, if not generates actual output file
+func checkActualVsExpectedOutputMatch(t *testing.T, testName, dirName, expectedOutputFileName, actualOutput, format string) {
+	actualOutputFileName := "actual_" + expectedOutputFileName
+	// read expected output file
+	expectedOutputFile := filepath.Join(getDirPathFromDirName(dirName), expectedOutputFileName)
+	expectedOutput, err := os.ReadFile(expectedOutputFile)
+	require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+	actualOutputFile := filepath.Join(getDirPathFromDirName(dirName), actualOutputFileName)
+	if string(expectedOutput) != actualOutput {
+		// generate actual output file		
+		fp, err := os.Create(actualOutputFile)
+		require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+		_, err = fp.WriteString(actualOutput)
+		require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+	}
+	require.Equal(t, string(expectedOutput), actualOutput, "output mismatch for %s, actual output file %q vs expected output file: %q", getDebugMsgWithTestNameAndFormat(testName, format),
+		actualOutputFile, expectedOutputFile)
+}
 
-var allFormats = []string{common.TextFormat, common.JSONFormat, common.CSVFormat, common.MDFormat, common.DOTFormat}
+// helping func - if focus workload is not empty append it to ConnlistAnalyzerOption list
+func appendFocusWorkloadOptIfRequired(focusWorkload string) []ConnlistAnalyzerOption {
+	analyzerOptions := []ConnlistAnalyzerOption{}
+	if focusWorkload != "" {
+		analyzerOptions = append(analyzerOptions, WithFocusWorkload(focusWorkload))
+	}
+	return analyzerOptions
+}
+
+// helping func - if the actual error/warning message does not contain the expected error, fail the test with relevant info
+func checkErrorContainment(t *testing.T, testName, expectedErrorMsg, actualErrMsg string, isErr bool) {
+	errType := "error"
+	if !isErr {
+		errType = "warning"
+	}
+	if !strings.Contains(actualErrMsg, expectedErrorMsg) {
+		t.Fatalf("%s message mismatch for test %q, actual: %q, expected contains: %q", errType, testName, actualErrMsg, expectedErrorMsg)
+	}
+	require.Contains(t, actualErrMsg, expectedErrorMsg, "test: %q", testName)
+}
```

Comment:
same here - no need for both `t.Fatalf` and `require.Contains` statements -- you can move the detailed str from the call of `Fatalf` to be the detailed str by the call to `require.Contains`

Reply:
eliminated all t.fatalf (my mistake : thought the msg in require would be printed always - so kept as test name and format only)